.github/workflows/archive-vscode-artifacts.yaml

```diff
@@ -19,18 +19,28 @@ jobs:
       - name: Setup antlr4
         uses: ./.github/actions/setup-antlr4
 
-      - name: Install dependencies with frozen lock file and generate parser
-        run: npm ci
+      - name: Setup pnpm
+        uses: pnpm/action-setup@v4
+
+      - name: Setup node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: 'pnpm'
+          registry-url: 'https://registry.npmjs.org'
```

Comment:
Could we maybe extract this to one of our custom actions (e.g. `setup-pnpm`) given it's used in a few places?

See the `./.github/actions/setup-antlr4` for inspiration

Reply:
yes for sure!